src/extensions/default/DebugCommands/main.js

```diff
@@ -257,6 +272,467 @@ define(function (require, exports, module) {
+            CommandManager.execute(Commands.FILE_CLOSE, {file: defaultPrefsFile, paneId: "first-pane"});
+        } else if (MainViewManager.findInWorkingSet("second-pane", defaultPrefsPath) >= 0) {
+            CommandManager.execute(Commands.FILE_CLOSE, {file: defaultPrefsFile, paneId: "second-pane"});
+        }
+
+        // Open the default preferences in the left pane in the read only mode.
+        CommandManager.execute(Commands.FILE_OPEN, { fullPath: defaultPrefsPath, paneId: "first-pane", options: { isReadOnly: true } });
+
+        if (currScheme.rows === 1 && currScheme.columns === 1) {
+            // Split layout is not active yet. Inititate the
+            // split view.
+            MainViewManager.setLayoutScheme(1, 2);
+        }
+
+
+        // Make sure the preference file is going to be opened in the second
+        // pane
+        if (MainViewManager.findInWorkingSet("first-pane", prefsPath) >= 0) {
+
+            MainViewManager._moveView("first-pane", "second-pane", file, 0, true);
+
+            // Now refresh the project tree by asking
+            // it to rebuild the UI.
+            WorkingSetView.refresh(true);
+        }
+
+        CommandManager.execute(Commands.FILE_OPEN, { fullPath: prefsPath, paneId: "second-pane"});
+
+    }
+
+    function _isSupportedPrefType(prefType) {
+        if (prefType !== undefined &&
+                (prefType === "number" ||
+                prefType === "boolean" ||
+                prefType === "string"  ||
+                prefType === "array"   ||
+                prefType === "object")) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    // This method tries to deduce the preference type
+    // based on various parameters like objects initial
+    // value, object type, object's type property.
+    function _getObjType(prefObj) {
+
+        var prefType = "undefined";
+
+        if (prefObj) {
+            // check the type parameter.
+            var _type = prefObj.type;
+            if (typeof (_type) !== "undefined") {
+                prefType = prefObj.type.toLowerCase();
+                // make sure the initial property's
+                // object type matches to that of 'type' propety.
+                if (prefObj.initial !== undefined) {
+
+                    if (Array.isArray(prefObj.initial)) {
+                        _type = "array";
+                    } else {
+                        var _initialType = typeof (prefObj.initial);
+                        _initialType = _initialType.toLowerCase();
+                        if (_type !== _initialType) {
+                            _type = _initialType;
+                        }
+                    }
+                }
+            }
+
+            if (_type) {
+                // preference object's type
+                // is defined. Check if that is valid or not.
+                prefType = _type;
+                if ((prefType !== "number"  &&
```

Comment:
OK! will do that. I have added `_isSupportedPrefType` recently and this code was there before.